whetstone/compiler/src/main/kotlin/com/freeletics/mad/whetstone/parser/ComposeParser.kt

```diff
@@ -0,0 +1,53 @@
+package com.freeletics.mad.whetstone.parser
+
+import com.freeletics.mad.whetstone.ComposeScreenData
+import com.freeletics.mad.whetstone.Navigation
+import com.freeletics.mad.whetstone.codegen.util.composeFqName
+import com.freeletics.mad.whetstone.codegen.util.composeNavDestinationFqName
+import com.freeletics.mad.whetstone.codegen.util.composeRootNavDestinationFqName
+import com.squareup.anvil.annotations.ExperimentalAnvilApi
+import com.squareup.anvil.compiler.internal.reference.AnnotatedReference
+
+@OptIn(ExperimentalAnvilApi::class)
+internal fun TopLevelFunctionReference.toComposeScreenData(): ComposeScreenData? {
+    val annotation = findAnnotation(composeFqName) ?: return null
+    val navigation = composeNavigation()
+
+    return ComposeScreenData(
+        baseName = name,
+        packageName = packageName,
+        scope = annotation.scope,
+        parentScope = annotation.parentScope,
+        stateMachine = annotation.stateMachine,
+        navigation = navigation,
+        navEntryData = navEntryData(navigation),
+        composableParameter = composeParameters
+    )
+}
+
+@OptIn(ExperimentalAnvilApi::class)
+private fun AnnotatedReference.composeNavigation(): Navigation.Compose? {
+    val navigation = findAnnotation(composeNavDestinationFqName)
+    if (navigation != null) {
+        val route = navigation.requireClassArgument("route", 0)
+        val destinationScope = navigation.requireClassArgument("destinationScope", 2)
```

Comment:
I didn't add utilities like `annotation.scope` for these because this method is going to completely change in #259 (same for the fragmentNavigation method in the next file)